src/Room/Messages.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useContext, useEffect, useState } from 'react'
+import { useQuery } from '@apollo/react-hooks'
+import moment from 'moment'
+
+import { WebsocketContext } from 'App'
+
+import { MESSAGES_QUERY, MessagesQuery, Message } from './graphql'
+
+const Messages: React.FC = () => {
+  const { data, loading } = useQuery<MessagesQuery['data'], MessagesQuery['vars']>(MESSAGES_QUERY)
+  const [messages, setMessages] = useState<Message[]>([])
+  const [newMessage, setNewMessage] = useState<Message | null>(null)
+
+  useEffect(() => {
+    if (!data) {
+      return
+    }
+
+    setMessages(data.messages)
+  }, [data])
+
+  useEffect(() => {
```

Comment:
I dunno exactly about this pattern, but it's the only way that I could get a new message from the `useEffect` webhook subscription to _append_ to existing messages rather than overwrite the whole list (because the `messages` variable does **not** seem to get updated automatically if I reference it within the `useEffect` hook itself.